test/integration/controllermanager/shootstate/shootstate_test.go

```diff
@@ -0,0 +1,239 @@
+		Expect(testClient.Status().Patch(ctx, shoot, patch)).To(Succeed())
+
+		By("Create ShootState")
+		shootState = &gardencorev1beta1.ShootState{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: testNamespace.Name,
+				Name:      shoot.Name,
+				Labels:    map[string]string{testID: testRunID},
+			},
+		}
+		Expect(testClient.Create(ctx, shootState)).To(Succeed())
+
+		DeferCleanup(func() {
+			By("Delete Shoot")
+			Expect(client.IgnoreNotFound(testClient.Delete(ctx, shoot))).To(Succeed())
+
+			By("Delete ShootState")
+			Expect(controllerutils.RemoveFinalizers(ctx, testClient, shootState, shootstate.FinalizerName)).To(Succeed())
+			Expect(client.IgnoreNotFound(testClient.Delete(ctx, shootState))).To(Succeed())
+
+			By("Ensure ShootState is gone")
+			Eventually(func() error {
+				return testClient.Get(ctx, client.ObjectKeyFromObject(shootState), shootState)
+			}).Should(BeNotFoundError())
+
+			By("Ensure Shoot is gone")
+			Eventually(func() error {
+				return testClient.Get(ctx, client.ObjectKeyFromObject(shoot), shoot)
+			}).Should(BeNotFoundError())
+
+		})
+	})
+
+	When("migrating Shoot to another Seed", func() {
+		var (
+			patch client.Patch
+		)
+
+		BeforeEach(func() {
+			By("Mark Shoot for migration")
+			patch = client.MergeFrom(shoot.DeepCopy())
+			shoot.Spec.SeedName = &targetSeedName
+			Expect(testClient.SubResource("binding").Patch(ctx, shoot, patch)).To(Succeed())
+		})
+
+		When("Shoot last operation is Migrate", func() {
+			BeforeEach(func() {
+				shoot.Status.LastOperation = &gardencorev1beta1.LastOperation{
+					Type:     gardencorev1beta1.LastOperationTypeMigrate,
+					State:    gardencorev1beta1.LastOperationStateProcessing,
+					Progress: 0,
+				}
+			})
+
+			It("should add finalizer if not present", func() {
+				By("Update the Shoot's last operation")
+				Expect(testClient.Status().Patch(ctx, shoot, patch)).To(Succeed())
+
+				By("Verify the finalizer is present")
+				Eventually(func(g Gomega) []string {
+					g.Expect(testClient.Get(ctx, client.ObjectKeyFromObject(shootState), shootState)).To(Succeed())
+					return shootState.Finalizers
+				}).Should(ConsistOf(shootstate.FinalizerName))
+			})
+
+			It("should not add/duplicate finalizer if already present", func() {
+				addFinalizer(shootState)
+
+				By("Update the Shoot's last operation")
+				Expect(testClient.Status().Patch(ctx, shoot, patch)).To(Succeed())
+
+				By("Verify the finalizer is present")
+				Consistently(func(g Gomega) []string {
+					g.Expect(testClient.Get(ctx, client.ObjectKeyFromObject(shootState), shootState)).To(Succeed())
+					return shootState.Finalizers
+				}).Should(Equal([]string{shootstate.FinalizerName}))
+			})
+		})
+
+		When("Shoot last operation is Restore", func() {
+			BeforeEach(func() {
+				shoot.Status.LastOperation = &gardencorev1beta1.LastOperation{
+					Type:     gardencorev1beta1.LastOperationTypeRestore,
+					State:    gardencorev1beta1.LastOperationStateProcessing,
+					Progress: 0,
```

Comment:
Remove this

Reply:
Thanks for pointing this redundant assignment, removed it.